bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/storage/ldb/ReadCache.java

```diff
@@ -73,7 +73,7 @@ public ReadCache(ByteBufAllocator allocator, long maxCacheSize, int maxSegmentSi
         cacheIndexes = new ArrayList<>();
 
         for (int i = 0; i < segmentsCount; i++) {
-            cacheSegments.add(Unpooled.directBuffer(segmentSize, segmentSize));
+            cacheSegments.add(Unpooled.buffer(segmentSize, segmentSize));
```

Comment:
This is not really coming from the allocator. Rather, the intention is to really get unpooled direct memory for the read cache.

Reply:
what is the side effect of being pooled?

Reply:
In this case that the memory is not returned to the pool (but rather the direct mem chunks will be just GCed when the handle object is GCed)

For this case, we allocate big chunks of mem (eg: 1GB each) and we keep using them indefinitely. Effectively this is a kind of "manual pooling" of the read and write cache memory.

There would be no advantage in pooling that memory again.

Reply:
but it doesn't hurt to use pooled, correct? from code maintenance perspective, isn't it clear to use allocator across the bookkeeper code base? If we need to use unpooled, we can use unpooled and document why do we need to use unpooled to make people know exact the reason behind the scene.

Reply:
> but it doesn't hurt to use pooled, correct? 

We'd have to at least release the buffers when the ReadCache/WriteCache are closed

> If we need to use unpooled, we can use unpooled and document why do we need to use unpooled to make people know exact the reason behind the scene. 

Sure, that would be better to be explicitely noted why it was that way

Reply:
> We'd have to at least release the buffers when the ReadCache/WriteCache are closed

shouldn't releasing buffers always be done when closing ReadCache/WriteCache. It is always be good to clean up  the resources even they are not used anymore. I didn't still get the point why there are benefits using unpooled buffers.

Reply:
Sure, I'm just saying that in this PR we need to add that release when switching from unpooled to pooled. 

> I didn't still get the point why there are benefits using unpooled buffers.

I would avoid the tracking of these buffers by the allocator. The overhead might be minimal, OTOH there's no advantage from the allocator either.

Reply:
> I would avoid the tracking of these buffers by the allocator. The overhead might be minimal, OTOH there's no advantage from the allocator either.

the intention is to make sure all the memory allocation is consistent across the bookie server and leverage whatever changes we made in our allocator, unless there are a very strong reason to not do so. 

since there is no advantages and also no disadvantages, why not just to make things consistent and easier to maintain?

Reply:
I'm not saying not to do it. just saying to add the release on close().

Reply:
If you are okay with using allocator, I will fix the release part.

Reply:
actually those buffers are already released on close